src/Employer/Employer.Web/Orchestrators/DeleteVacancyOrchestrator.cs

```diff
@@ -17,16 +19,19 @@ public DeleteVacancyOrchestrator(IEmployerVacancyClient client)
             _client = client;
         }
 
-        public async Task<DeleteViewModel> GetDeleteViewModelAsync(Guid vacancyId)
+        public async Task<DeleteViewModel> GetDeleteViewModelAsync(VacancyRouteModel vrm)
         {
-            var vacancy = await _client.GetVacancyAsync(vacancyId);
+            var vacancy = await _client.GetVacancyAsync(vrm.VacancyId);
+            
+            if (!vacancy.EmployerAccountId.Equals(vrm.EmployerAccountId, StringComparison.OrdinalIgnoreCase))
+                throw new AuthorisationException(string.Format(ExceptionMessages.VacancyUnauthorisedAccess, vrm.EmployerAccountId, vacancy.EmployerAccountId, vacancy.Title, vacancy.Id));
```

Comment:
I would prefer not to put this in an extension method. I don't like the idea of augmenting the Vacancy entity with this behavior or being concerned with throwing an exception. Leaving it where it is it keeps it open for the orchestrator to apply the check rather than just call it. Also it would be odd when using the Vacancy entity in other consuming apps where this function would not be there if it is an extension in Employer.Web.

Reply:
OK. There's a lot of repetition and the different ways of working out authorisation is not sitting well with me.  Seems too easy to forget to apply the check.  Wondering if we could have a wrapper class around VacancyClient which has a function `GetVacancyAsync(VacancyId, EmployerAccountId)`. So you can only get a vacancy if you specify the account id too.

Reply:
I agree about the repetition completely and did try to reduce it. It is still in more than 2 places so it should be put in a common place. Forgetting to call it is also a concern I have had. However I can't think of a better solution and out of the ones suggested, the one in this PR is what I am content with. I have no issue with changing it to a way that the rest of the team agrees with. I will bring it up on Monday.

Reply:
Might be a controversial one but could we do something like the following: [https://gist.github.com/leewadhams/6d4b8c2d645b5eef4b68eb32690351c8](https://gist.github.com/leewadhams/6d4b8c2d645b5eef4b68eb32690351c8)

Reply:
Thanks Lee, I have decided to just plonk the call in a regular static method as then it is in one place and we can make a call later on.